src/test/kotlin/io/sip3/salto/ce/socket/ManagementSocketTest.kt

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018-2019 SIP3.IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.salto.ce.socket
+
+import io.sip3.commons.domain.SdpSession
+import io.sip3.commons.vertx.test.VertxTest
+import io.sip3.salto.ce.MongoExtension
+import io.sip3.salto.ce.RoutesCE
+import io.sip3.salto.ce.USE_LOCAL_CODEC
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.mongo.MongoClient
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.net.ServerSocket
+import java.util.*
+
+@ExtendWith(MongoExtension::class)
+class ManagementSocketTest : VertxTest() {
+
+    companion object {
+
+        private val host = JsonObject().apply {
+            put("name", "sbc1")
+            put("sip", arrayListOf("10.10.10.10", "10.10.20.10:5060"))
+        }
+    }
+
+    private lateinit var config: JsonObject
+    private var localPort = -1
+    private var remotePort = -1
+    private val registerMessage = JsonObject().apply {
+        put("type", ManagementSocket.TYPE_REGISTER)
+        put("payload", JsonObject().apply {
+            put("name", UUID.randomUUID().toString())
+            put("host", host)
+        })
+    }
+
+    @BeforeEach
+    fun init() {
+        val localSocket = ServerSocket(0)
```

Comment:
I guess that we use this code to define free ports? Maybe it makes sense to google for extension like this or to move this code to junit extension?